File: Estruturas de Dados/Avaliações/Trabalho02/clima.py
class Clima:
    def __init__(self, ano, mes, temperatura, precipitacao):
        self.ano = ano
        self.mes = mes
        self.temperatura = temperatura
        self.precipitacao = precipitacao
    
    def __str__(self):
        return f"Ano: {self.ano}, Mês: {self.mes}, Temperatura: {self.temperatura}°C, Precipitação: {self.precipitacao}mm"
    
    def __eq__(self, other):
        if not isinstance(other, Clima):
            return False
        return self.ano == other.ano and self.mes == other.mes
    
    def get_valor_temperatura(self):
        temp = self.temperatura.lower()
        mapping = {
            "frio": 1,
            "morno": 2,
            "quente": 3
        }

        return mapping.get(temp, 0)  # Retorna 0 se a temperatura não for reconhecida

File: Estruturas de Dados/Avaliações/Trabalho 2/Clima.java
public class Clima{
    public String ano;
    public String mes;
    public String temperatura;
    public String precipitacao;

    public Clima(String ano, String mes, String temperatura, String precipitacao) {
        this.ano = ano;
        this.mes = mes;
        this.temperatura = temperatura;
        this.precipitacao = precipitacao;
    }

    @Override
    public String toString() {
        return "Dados climaticos(Ano: " + this.ano + ", Mês: " + this.mes + ", Temperatura: " + this.temperatura + ", Precipitação: " + this.precipitacao + ")";
    }
    //reescrevemos toString para exibir um objeeto completo da classe 
    @Override
    public boolean equals(Object obj) {
        Clima c = (Clima) obj;
        return (this.ano.equals(c.ano) && this.mes.equals(c.mes));
        //estamos dizendo quem são as chaves primárias (aquelas que não podem se repetir)
        //reescrevemos o metodo equals para definir os atributos identificadores
    }

    // método para converter a temperatura em um valor numérico

    public int getValorTemperatura(){
        String temp = this.temperatura.toLowerCase();
        if(temp.equals("frio")){
            return 1;
        } else if(temp.equals("morno")){
            return 2;
        } else if(temp.equals("quente")){
            return 3;
        } else {
            return 0; // valor padrão para casos não reconhecidos
        }
    }
    
}
